xpublish/rest.py

```diff
@@ -52,18 +52,18 @@ def __init__(self, xarray_obj):
 
     def _get_zmetadata(self):
         """ helper method to create consolidated zmetadata dictionary """
-        zmeta = {"zarr_consolidated_format": zarr_consolidated_format, "metadata": {}}
-        zmeta["metadata"][group_meta_key] = {"zarr_format": zarr_format}
-        zmeta["metadata"][attrs_key] = self._get_zattrs()
+        zmeta = {'zarr_consolidated_format': zarr_consolidated_format, 'metadata': {}}
+        zmeta['metadata'][group_meta_key] = {'zarr_format': zarr_format}
+        zmeta['metadata'][attrs_key] = self._get_zattrs()
 
         for key, da in self._obj.variables.items():
             # encode variable
             encoded_da = encode_zarr_variable(da)
```

Comment:
I don't know whether this is a bug or the expected behavior, but when I pass `time_bounds(time, d2)` (which is a dask array) to `encode_zarr_variable()` function from xarray, I ended up getting a numpy array:
 

```python
In [24]: from xarray.backends.zarr import encode_zarr_variable                                                     

In [25]: encoded = encode_zarr_variable(ds2.time_bounds)                                                           

In [26]: ds2.time_bounds                                                                                           
Out[26]: 
<xarray.DataArray 'time_bounds' (time: 36, d2: 2)>
dask.array<xarray-time_bounds, shape=(36, 2), dtype=datetime64[ns], chunksize=(6, 2), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) datetime64[ns] 2018-01-16T12:00:00 ... 2020-12-16T12:00:00
Dimensions without coordinates: d2

In [27]: encoded                                                                                                   
Out[27]: 
<xarray.Variable (time: 36, d2: 2)>
array([[   0,   31],
       [  31,   59],
       [  59,   90],
       [  90,  120],
       [ 120,  151],
       [ 151,  181],
       [ 181,  212],
       [ 212,  243],
       [ 243,  273],
       [ 273,  304],
       [ 304,  334],
       [ 334,  365],
       [ 365,  396],
       [ 396,  424],
       [ 424,  455],
       [ 455,  485],
       [ 485,  516],
       [ 516,  546],
       [ 546,  577],
       [ 577,  608],
       [ 608,  638],
       [ 638,  669],
       [ 669,  699],
       [ 699,  730],
       [ 730,  761],
       [ 761,  790],
       [ 790,  821],
       [ 821,  851],
       [ 851,  882],
       [ 882,  912],
       [ 912,  943],
       [ 943,  974],
       [ 974, 1004],
       [1004, 1035],
       [1035, 1065],
       [1065, 1096]])
Attributes:
    units:     days since 2018-01-01 00:00:00
    calendar:  proleptic_gregorian
```

This issue appears to be the reason behind the latest test failures because at the following lines:

https://github.com/jhamman/xpublish/blob/cbaebbaf8a1f4332f207210fdbcebc2a83547043/xpublish/rest.py#L316-L317

`var_chunks=(36, 2)` and `tuple(meta['chunks']) = (10, 2)`

Reply:
Nice work tracking this down. I'm not 100% sure but I think this line is the culprit in xarray: https://github.com/pydata/xarray/blob/01462d65c7213e5e1cddf36492c6a34a7e53ce55/xarray/coding/times.py#L353

Perhaps we xfail datasets with datetime variables for now?

Reply:
Done!

Reply:
I just opened an issue upstream @ https://github.com/pydata/xarray/issues/3834

Reply:
Thanks!